### PR TITLE
cephfs: add file xattr related functions

### PR DESCRIPTION
**Pull request by phlogistonjohn**
Add functions wrapping ceph_fsetxattr, ceph_fgetxattr,  ceph_flistxattr, and ceph_fremovexattr. Tests for all are included. See final commit for an interesting wrinkle in the behavior in ceph.

Fixes: #247 

## Checklist
- [x] Added tests for features and functional changes
- [x] Public functions and types are documented
- [x] Standard formatting is applied to Go code

### CODE REVIEW

**Comment by phlogistonjohn** [comment]
I put the DNM prefix on to give the ceph team a little time to reply to the issue I filed. If we get quick turnaround on it I can update the code/tests as needed. If not, we can merge this and then just update the docs/tests later.

File: cephfs/file_xattr.go [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+	// XattrCreate specified that set-xattr calls only set new xattrs.
+	XattrCreate = XattrFlags(C.XATTR_CREATE)
+	// XattrReplace specified that set-xattr calls only replace existing xattr
+	// values.
+	XattrReplace = XattrFlags(C.XATTR_REPLACE)
+)
+
+// SetXattr sets an extended attribute on the open file.
+//
+// NOTE: Attempting o set an xattr value with an empty value may cause
```
Comment (nixpanic) [inline cephfs/file_xattr.go:37]: "o"?

File: cephfs/file_xattr.go [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+// the xattr to be unset. Please refer to https://tracker.ceph.com/issues/46084
```
Comment (nixpanic) [inline cephfs/file_xattr.go:38]: There is some progress here. But I doubt all users would update to a version that behaves more correctly. So, assume that broken ceph versions will be used (probably do/not handle that with testing). Documenting the behaviour is sufficient for me.

File: cephfs/file_xattr.go [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+// GetXattr gets an extended attribute from the open file.
+//
+// Implements:
+//  int ceph_fgetxattr(struct ceph_mount_info *cmount, int fd, const char *name,
+//                     void *value, size_t size);
+func (f *File) GetXattr(name string) ([]byte, error) {
+	if err := f.validate(); err != nil {
+		return nil, err
+	}
+	cName := C.CString(name)
```
Comment (nixpanic) [inline cephfs/file_xattr.go:73]: need to ckeck if `name == ""`?

File: cephfs/file_xattr.go (hunk 1 above)
Comment (phlogistonjohn) [reply cephfs/file_xattr.go:37]: The "t" in "to" must have ran away! ;-)

File: cephfs/file_xattr.go (hunk 3 above)
Comment (phlogistonjohn) [reply cephfs/file_xattr.go:73]: Sounds reasonable, will do.

**Comment by phlogistonjohn** [comment]
I've removed the DNM label, as I think the documentation approach is good enough. Please take a look at said doc comment and let me know if you think it needs expanding or if it is sufficient in the current form.

File: cephfs/file_xattr.go [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+	"unsafe"
+
+	"github.com/ceph/go-ceph/internal/retry"
+)
+
+// XattrFlags are used to control the behavior of set-xattr calls.
+type XattrFlags int
+
+const (
+	// XattrDefault specifies that set-xattr calls use the default behvior of
```
Comment (agarwal-mudit) [inline cephfs/file_xattr.go:25]: nit: behavior

File: cephfs/file_xattr.go [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+	// creating or updating an xattr.
+	XattrDefault = XattrFlags(0)
+	// XattrCreate specified that set-xattr calls only set new xattrs.
```
Comment (agarwal-mudit) [inline cephfs/file_xattr.go:28]: nit: specifies

File: cephfs/file_xattr.go [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+	XattrCreate = XattrFlags(C.XATTR_CREATE)
+	// XattrReplace specified that set-xattr calls only replace existing xattr
```
Comment (agarwal-mudit) [inline cephfs/file_xattr.go:30]: nit: specifies

File: cephfs/file_xattr.go [hunk 7]
```diff
@@ -97,3 +98,43 @@ func (f *File) GetXattr(name string) ([]byte, error) {
...
+			(*C.char)(unsafe.Pointer(&buf[0])),
+			C.size_t(size))
+		err = getErrorIfNegative(ret)
+		return retry.DoubleSize.If(err == errRange)
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	names := make([]string, 0)
```
Comment (agarwal-mudit) [inline cephfs/file_xattr.go:135]: Shouldn't we wait for #313 to get merged?

Comment (phlogistonjohn) [reply cephfs/file_xattr.go:135]: This prompted me to work on that one but I don't think it matters which one lands first. If it this one, I will update #313 to include these functions. If #313 lands first, I will update this PR to make use of it.

File: cephfs/file_xattr.go (hunk 4 above)
Comment (phlogistonjohn) [reply cephfs/file_xattr.go]: This and the other spelling mistakes are fixed.

**Comment by phlogistonjohn** [comment]
@nixpanic can you please (re-)review this and give an OK (or not)?

File: cephfs/file_xattr.go (hunk 7 above)
Comment (agarwal-mudit) [reply cephfs/file_xattr.go:135]: #313 is merged now.

Comment (phlogistonjohn) [reply cephfs/file_xattr.go]: I've pushed a follow up patch to the PR, PTAL.

**Pull request merged**